File: src/rules/jwt.py
import jwt


def decode_no_verify(token, key):
    # ruleid: clipeus-jwt-verify-disabled-python
    return jwt.decode(token, key, verify=False)


def decode_options_off(token, key):
    # ruleid: clipeus-jwt-verify-disabled-python
    return jwt.decode(token, key, options={"verify_signature": False})


def decode_alg_none(token, key):
    # ruleid: clipeus-jwt-algorithm-none-python
    return jwt.decode(token, key, algorithms=["none"])


def decode_safe(token, key):
    # ok: clipeus-jwt-verify-disabled-python
    # ok: clipeus-jwt-algorithm-none-python
    return jwt.decode(token, key, algorithms=["RS256"])

File: src/rules/jwt.js
const jwt = require('jsonwebtoken');

function verifyNone(token) {
  // ruleid: patronus-jwt-algorithm-none
  return jwt.verify(token, key, { algorithms: ['none'] });
}

function signNone(payload) {
  // ruleid: patronus-jwt-algorithm-none
  return jwt.sign(payload, key, { algorithm: 'none' });
}

function signHardcoded(payload) {
  // ruleid: patronus-jwt-hardcoded-secret
  return jwt.sign(payload, 'super-secret-key', { expiresIn: '1h' });
}

function verifyHardcoded(token) {
  // ruleid: patronus-jwt-hardcoded-secret
  return jwt.verify(token, 'super-secret-key');
}

function signNoExpiry(payload) {
  // ruleid: patronus-jwt-missing-expiration
  return jwt.sign(payload, process.env.JWT_SECRET);
}

function signSafe(payload) {
  // ok: patronus-jwt-missing-expiration
  // ok: patronus-jwt-hardcoded-secret
  return jwt.sign(payload, process.env.JWT_SECRET, { expiresIn: '15m', algorithm: 'HS256' });
}

function verifySafe(token) {
  // ok: patronus-jwt-algorithm-none
  return jwt.verify(token, publicKey, { algorithms: ['RS256'] });
}

module.exports = { verifyNone, signNone, signHardcoded, verifyHardcoded, signNoExpiry, signSafe, verifySafe };
